openslides/core/static/js/core/base.js

```diff
@@ -493,22 +499,46 @@ angular.module('OpenSlidesApp.core', [
 
 .directive('templateHook', [
     '$compile',
+    '$http',
+    '$q',
     'templateHooks',
-    function ($compile, templateHooks) {
+    function ($compile, $http, $q, templateHooks) {
         return {
             restrict: 'E',
             template: '',
             link: function (scope, iElement, iAttr) {
                 var hooks = templateHooks.hooks[iAttr.hookName];
-                var html;
                 if (hooks) {
-                    html = hooks.map(function (hook) {
-                        return '<div>' + hook.template + '</div>';
-                    }).join('');
-                } else {
-                    html = '';
+                    var promises = _.map(hooks, function (hook) {
+                        // Either a template (html given as string) or a templateUrl has
+                        // to be given. If a scope is provided, the schope of this templateHook
+                        // is populated with the given functions/values.
+                        if (hook.template) {
+                            return $q(function (resolve) {
+                                resolve('<div>' + hook.template + '</div>');
+                            });
+                        } else {
+                            return $q(function (resolve) {
```

Comment:
This wrapper around $http is not necessary I think. You can just return the $http.get().then(). Inside the then() function you may just return success.data.